firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -203,92 +215,92 @@ public synchronized void unregisterActivityLifecycleCallbacks() {
     if (!isRegisteredForLifecycleCallbacks) {
       return;
     }
+    ProcessLifecycleOwner.get().getLifecycle().removeObserver(this);
     ((Application) appContext).unregisterActivityLifecycleCallbacks(this);
     isRegisteredForLifecycleCallbacks = false;
   }
 
   /**
-   * Gets the timetamp that marks the beginning of app start, currently defined as the beginning of
-   * BIND_APPLICATION. Fallback to class-load time of {@link StartupTime} when API < 24.
+   * Gets the timestamp that marks the beginning of app start, defined as the beginning of
+   * BIND_APPLICATION, when the forked process is about to start loading the app's resources and
+   * classes. Fallback to class-load time of a Firebase class for compatibility below API 24.
+   *
+   * @return {@link Timer} at the beginning of app start by Firebase-Performance definition.
+   */
+  private @NonNull Timer getStartTimerCompat() {
+    // Preferred: Android system API provides BIND_APPLICATION time
+    if (processStartTime != null) {
+      return processStartTime;
+    }
+    // Fallback: static initializer time (during class-load) of a Firebase class
+    return getClassLoadTimeCompat();
+  }
+
+  /**
+   * Timestamp during class load. This timestamp is captured in a static initializer during class
+   * loading of a particular Firebase class, the order of which CANNOT be guaranteed to be prior to
+   * all other classes of an app, nor prior to resource-loading of an app. Thus this timestamp is
+   * NOT PREFERRED to be used as starting-point of app-start.
    *
-   * @return {@link Timer} at the beginning of app start by Fireperf definition.
+   * @return {@link Timer} captured by static-initializer during class-loading of a Firebase class.
    */
-  private static Timer getStartTimer() {
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
-      return Timer.ofElapsedRealtime(
-          Process.getStartElapsedRealtime(), Process.getStartUptimeMillis());
+  private @NonNull Timer getClassLoadTimeCompat() {
+    // Prefered: static-initializer time of the 1st Firebase class during init
+    if (firebaseClassLoadTime != null) {
+      return firebaseClassLoadTime;
     }
-    return firebaseStartupTime;
+    // Fallback: static-initializer time of the current class
+    return PERF_CLASS_LOAD_TIME;
```

Comment:
No action needed. Just to confirm, we aren't concerned with which class load time to use right? Potentially because the values should still be really close.
It doesn't look like there is a mechanism to determine which one was used.

firebaseClassLoadTime vs PERF_CLASS_LOAD_TIME